Source/Applications/SystemCenter/wwwroot/Scripts/TSX/SystemCenter/Asset/AssetSelect.tsx

```diff
@@ -49,15 +50,27 @@ export default function AssetSelect(props: IProps) {
     const assetTypeStatus = useAppSelector(AssetTypeSlice.Status);
     const dispatch = useAppDispatch();
 
-    const lineSegmentFilter: Search.IFilter<SystemCenter.Types.DetailedAsset> =
-    {
+    const lineSegmentFilter: Search.IFilter<SystemCenter.Types.DetailedAsset> = {
         FieldName: 'AssetType',
         SearchText: 'LineSegment',
         Operator: '<>',
         Type: 'string',
         IsPivotColumn: false
     };
 
+    const locationFilter: Search.IFilter<SystemCenter.Types.DetailedAsset> = {
+        FieldName: "Location",
+        SearchText: `${props.LocationKey}`,
+        Operator: "LIKE",
+        Type: "string",
+        IsPivotColumn: false
+    }
+
+    React.useEffect(() => {
+        if (props.LocationKey != undefined && props.StorageID != undefined)
+            localStorage.setItem(`${props.StorageID}.Filters`, JSON.stringify([locationFilter]));
+    }, [props.LocationKey])
```

Comment:
so what happens if there are already other filters in storage?

Reply:
updated to check for this